### PR TITLE
Reintroduce user cache

### LINKED ISSUE #16942: "Invalid image" when trying to upload user pic on oC 8.1Beta
### Steps to reproduce
1. go to Personal settings
2. Choose to upload and pick a PNG/png, JPG/jpg file.
3. "Invalid image" is all it says

### Expected behaviour
It should show the nice image you picked...

### Actual behaviour
Invalid image:
statistics june 15 2016
This is with jpg and JPG images, png and PNG's, in both Chromium and Firefox.

A different way of doing things is uploading an image separately (works) and then choosing it as avatar. That gives a different problem:
statistics june 15 2016
statistics june 15 2016
statistics june 15 2016
statistics june 15 2016

(pictures is not changed at all)

### Server configuration
**Operating system**:
openSUSE Tumbleweed
**Web server:**
Name: apache2
Version: 2.4.12-7.1
Arch: x86_64

**Database:**
Name: sqlite3
Version: 3.8.10.2-1.1

**PHP version:**
Name: php5
Version: 5.6.9-1.1

**ownCloud version:** (see ownCloud admin page)
8.1 beta 2 (daily) Build:2015-06-15T03:38:05+00:00 fe9b28a800514315eba17ac5da84b8b93abd9ffc

**Updated from an older ownCloud or fresh install:**
Fresh install (extracted tarball - gave user name - nothing else)

**List of activated apps:**
Default + external storage

**The content of config/config.php:**

```
Insert your config.php content here

<?php
$CONFIG = array (
  'instanceid' => 'oc7jqt7p8otl',
  'trusted_domains' =>
  array (
    0 => 'localhost',
  ),
  'datadirectory' => '/srv/www/htdocs/owncloud/data',
  'overwrite.cli.url' => 'http://localhost/owncloud',
  'dbtype' => 'sqlite3',
  'version' => '8.1.0.6',
  'logtimezone' => 'UTC',
  'installed' => true,
  'loglevel' => 0,
);
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
local

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Firefox, Chromium

**Operating system:**
openSUSE

#### Web server error log
```
[Mon Jun 15 09:28:29.146106 2015] [ssl:warn] [pid 6958] AH01909: popcorn:443:0 server certificate does NOT include an ID which matches the server name
[Mon Jun 15 09:29:17.719368 2015] [php5:error] [pid 7498] [client ::1:47872] creating sqlite db, referer: https://localhost/owncloud/index.php/settings/admin
[Mon Jun 15 09:51:22.755972 2015] [ssl:warn] [pid 6958] AH01909: popcorn:443:0 server certificate does NOT include an ID which matches the server name
```

#### ownCloud log (data/owncloud.log)
```
{"reqId":"BuVtInrj9Bq0HBvGX7aR","remoteAddr":"::1","app":"PHP","message":"touch(): Unable to create file \/srv\/www\/htdocs\/owncloud\/data\/owncloud.log because No such file or directory at \/srv\/www\/htdocs\/owncloud\/lib\/private\/log\/owncloud.php#60","level":3,"time":"2015-06-15T08:52:57+00:00"}
{"reqId":"4YmBz6YpgS6\/73thpdVu","remoteAddr":"::1","app":"PHP","message":"stat(): stat failed for \/media\/bigbtrfs\/Pictures\/ at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/storage\/local.php#122","level":3,"time":"2015-06-15T09:04:23+00:00"}
{"reqId":"9z5ciWynMWGLNlhZNbml","remoteAddr":"::1","app":"PHP","message":"stat(): stat failed for \/media\/bigbtrfs\/Pictures\/ at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/storage\/local.php#122","level":3,"time":"2015-06-15T09:04:29+00:00"}
{"reqId":"duGE4Y030IonCYR+EMMG","remoteAddr":"::1","app":"PHP","message":"stat(): stat failed for \/media\/bigbtrfs\/pictures\/ at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/storage\/local.php#122","level":3,"time":"2015-06-15T09:04:35+00:00"}
{"reqId":"fxH1pgL0OgFeFF1ZodHO","remoteAddr":"::1","app":"PHP","message":"stat(): stat failed for \/media\/bigbtrfs\/Pictures\/ at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/storage\/local.php#122","level":3,"time":"2015-06-15T09:05:36+00:00"}
{"reqId":"L+K1ozH\/z+NengIGZDHS","remoteAddr":"::1","app":"PHP","message":"Array to string conversion at \/srv\/www\/htdocs\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-06-15T09:53:27+00:00"}
{"reqId":"nEqZNap4c6cHWIfOEwZ7","remoteAddr":"::1","app":"PHP","message":"Array to string conversion at \/srv\/www\/htdocs\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-06-15T09:53:39+00:00"}
{"reqId":"rOwaLQ\/06cF69durtR+H","remoteAddr":"::1","app":"PHP","message":"Array to string conversion at \/srv\/www\/htdocs\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-06-15T12:38:15+00:00"}
{"reqId":"cPJeVFliejkY1NJbXQE4","remoteAddr":"::1","app":"PHP","message":"Array to string conversion at \/srv\/www\/htdocs\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-06-15T14:58:49+00:00"}
{"reqId":"eTg\/H0T3bgLqkrA9On2m","remoteAddr":"::1","app":"PHP","message":"Array to string conversion at \/srv\/www\/htdocs\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-06-15T15:02:48+00:00","method":"GET","url":"\/owncloud\/index.php\/settings\/admin"}
{"reqId":"1nqnzNZB1VIryHhzFo6u","remoteAddr":"::1","app":"PHP","message":"Array to string conversion at \/srv\/www\/htdocs\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-06-15T15:03:21+00:00","method":"POST","url":"\/owncloud\/index.php\/settings\/admin"}
{"reqId":"yfG1koR18lrhIv1gYyNQ","remoteAddr":"::1","app":"no app in context","message":"CSRF check failed","level":0,"time":"2015-06-15T15:04:23+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"U8N8RGO8uzgcHdw+EOX2","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Image is not a JPEG.","level":0,"time":"2015-06-15T15:04:41+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"U8N8RGO8uzgcHdw+EOX2","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:04:41+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"kII0p1guOialsxEQDSNH","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Image is not a JPEG.","level":0,"time":"2015-06-15T15:07:11+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"kII0p1guOialsxEQDSNH","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:07:11+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"yzRtIHEcwYi9mo5QdWU5","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Image is not a JPEG.","level":0,"time":"2015-06-15T15:07:58+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"yzRtIHEcwYi9mo5QdWU5","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:07:58+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"jG5CKuR3rZ1ztpA+iWZE","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Image is not a JPEG.","level":0,"time":"2015-06-15T15:28:15+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"jG5CKuR3rZ1ztpA+iWZE","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:28:15+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"Ng8oADtEskxUQ8yu7bG3","remoteAddr":"","app":"DeleteOrphanedSharesJob","message":"0 orphaned share(s) deleted","level":0,"time":"2015-06-15T15:29:32+00:00","method":"--","url":"--"}
{"reqId":"ggtBZrhur7GkGIyp6iKq","remoteAddr":"::1","app":"PHP","message":"Automatically populating $HTTP_RAW_POST_DATA is deprecated and will be removed in a future version. To avoid this warning set 'always_populate_raw_post_data' to '-1' in php.ini and use the php:\/\/input stream instead. at Unknown#0","level":3,"time":"2015-06-15T15:29:44+00:00","method":"POST","url":"\/owncloud\/index.php\/displaynames"}
{"reqId":"5CPxFsdwRta3JilnSVhA","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Image is not a JPEG.","level":0,"time":"2015-06-15T15:34:02+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"5CPxFsdwRta3JilnSVhA","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:34:02+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"evhvFr\/aYW1cdr4N4EeV","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Image is not a JPEG.","level":0,"time":"2015-06-15T15:34:11+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"evhvFr\/aYW1cdr4N4EeV","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:34:11+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"I7sW6firqG+zpUjqOtvx","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Image is not a JPEG.","level":0,"time":"2015-06-15T15:34:31+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"I7sW6firqG+zpUjqOtvx","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:34:31+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"xG6ZXA8cNC9qa+YEETWg","remoteAddr":"::1","app":"core","message":"Generating preview for \"\/user pictures\/Guy Fawkes.png\" with \"OC\\Preview\\PNG\"","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy+Fawkes.png"}
{"reqId":"xG6ZXA8cNC9qa+YEETWg","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Image is not a JPEG.","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy+Fawkes.png"}
{"reqId":"xG6ZXA8cNC9qa+YEETWg","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy+Fawkes.png"}
{"reqId":"PI\/orGf\/14U\/If4kFgUh","remoteAddr":"::1","app":"core","message":"Generating preview for \"\/user pictures\/Guy_Fawkes_Mask_Image.jpg\" with \"OC\\Preview\\JPEG\"","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy_Fawkes_Mask_Image.jpg"}
{"reqId":"PI\/orGf\/14U\/If4kFgUh","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy_Fawkes_Mask_Image.jpg"}
{"reqId":"xG6ZXA8cNC9qa+YEETWg","remoteAddr":"::1","app":"files_locking","message":"INFO: Write lock requested for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/25\/500-500-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy+Fawkes.png"}
{"reqId":"xG6ZXA8cNC9qa+YEETWg","remoteAddr":"::1","app":"files_locking","message":"PASS: Write lock succeeded on \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/25\/500-500-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy+Fawkes.png"}
{"reqId":"xG6ZXA8cNC9qa+YEETWg","remoteAddr":"::1","app":"files_locking","message":"INFO: Incrementing lock type 2 count for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/25\/500-500-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy+Fawkes.png"}
{"reqId":"PI\/orGf\/14U\/If4kFgUh","remoteAddr":"::1","app":"files_locking","message":"INFO: Write lock requested for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/22\/600-600-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy_Fawkes_Mask_Image.jpg"}
{"reqId":"PI\/orGf\/14U\/If4kFgUh","remoteAddr":"::1","app":"files_locking","message":"PASS: Write lock succeeded on \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/22\/600-600-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy_Fawkes_Mask_Image.jpg"}
{"reqId":"PI\/orGf\/14U\/If4kFgUh","remoteAddr":"::1","app":"files_locking","message":"INFO: Incrementing lock type 2 count for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/22\/600-600-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy_Fawkes_Mask_Image.jpg"}
{"reqId":"\/VqFG+LeEl+\/Zj0qFEMA","remoteAddr":"::1","app":"core","message":"Generating preview for \"\/user pictures\/mad_people.jpg\" with \"OC\\Preview\\JPEG\"","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fmad_people.jpg"}
{"reqId":"\/VqFG+LeEl+\/Zj0qFEMA","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fmad_people.jpg"}
{"reqId":"PI\/orGf\/14U\/If4kFgUh","remoteAddr":"::1","app":"files_locking","message":"INFO: Write lock requested for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/22\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy_Fawkes_Mask_Image.jpg"}
{"reqId":"PI\/orGf\/14U\/If4kFgUh","remoteAddr":"::1","app":"files_locking","message":"PASS: Write lock succeeded on \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/22\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy_Fawkes_Mask_Image.jpg"}
{"reqId":"PI\/orGf\/14U\/If4kFgUh","remoteAddr":"::1","app":"files_locking","message":"INFO: Incrementing lock type 2 count for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/22\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy_Fawkes_Mask_Image.jpg"}
{"reqId":"\/VqFG+LeEl+\/Zj0qFEMA","remoteAddr":"::1","app":"files_locking","message":"INFO: Write lock requested for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/24\/400-355-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fmad_people.jpg"}
{"reqId":"\/VqFG+LeEl+\/Zj0qFEMA","remoteAddr":"::1","app":"files_locking","message":"PASS: Write lock succeeded on \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/24\/400-355-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fmad_people.jpg"}
{"reqId":"\/VqFG+LeEl+\/Zj0qFEMA","remoteAddr":"::1","app":"files_locking","message":"INFO: Incrementing lock type 2 count for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/24\/400-355-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fmad_people.jpg"}
{"reqId":"lIvicCDJeC66lnGietzd","remoteAddr":"::1","app":"core","message":"Generating preview for \"\/user pictures\/thinkwrong.png\" with \"OC\\Preview\\PNG\"","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fthinkwrong.png"}
{"reqId":"\/VqFG+LeEl+\/Zj0qFEMA","remoteAddr":"::1","app":"files_locking","message":"INFO: Write lock requested for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/24\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fmad_people.jpg"}
{"reqId":"\/VqFG+LeEl+\/Zj0qFEMA","remoteAddr":"::1","app":"files_locking","message":"PASS: Write lock succeeded on \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/24\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fmad_people.jpg"}
{"reqId":"\/VqFG+LeEl+\/Zj0qFEMA","remoteAddr":"::1","app":"files_locking","message":"INFO: Incrementing lock type 2 count for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/24\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fmad_people.jpg"}
{"reqId":"lIvicCDJeC66lnGietzd","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Image is not a JPEG.","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fthinkwrong.png"}
{"reqId":"lIvicCDJeC66lnGietzd","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fthinkwrong.png"}
{"reqId":"xG6ZXA8cNC9qa+YEETWg","remoteAddr":"::1","app":"files_locking","message":"INFO: Write lock requested for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/25\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy+Fawkes.png"}
{"reqId":"xG6ZXA8cNC9qa+YEETWg","remoteAddr":"::1","app":"files_locking","message":"PASS: Write lock succeeded on \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/25\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy+Fawkes.png"}
{"reqId":"xG6ZXA8cNC9qa+YEETWg","remoteAddr":"::1","app":"files_locking","message":"INFO: Incrementing lock type 2 count for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/25\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2FGuy+Fawkes.png"}
{"reqId":"lIvicCDJeC66lnGietzd","remoteAddr":"::1","app":"files_locking","message":"INFO: Write lock requested for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/23\/400-400-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fthinkwrong.png"}
{"reqId":"lIvicCDJeC66lnGietzd","remoteAddr":"::1","app":"files_locking","message":"PASS: Write lock succeeded on \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/23\/400-400-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fthinkwrong.png"}
{"reqId":"lIvicCDJeC66lnGietzd","remoteAddr":"::1","app":"files_locking","message":"INFO: Incrementing lock type 2 count for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/23\/400-400-max.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fthinkwrong.png"}
{"reqId":"lIvicCDJeC66lnGietzd","remoteAddr":"::1","app":"files_locking","message":"INFO: Write lock requested for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/23\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fthinkwrong.png"}
{"reqId":"lIvicCDJeC66lnGietzd","remoteAddr":"::1","app":"files_locking","message":"PASS: Write lock succeeded on \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/23\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fthinkwrong.png"}
{"reqId":"lIvicCDJeC66lnGietzd","remoteAddr":"::1","app":"files_locking","message":"INFO: Incrementing lock type 2 count for \/srv\/www\/htdocs\/owncloud\/data\/snowden\/thumbnails\/23\/36-36.png","level":0,"time":"2015-06-15T15:35:18+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/preview.png?file=%2Fuser+pictures%2Fthinkwrong.png"}
{"reqId":"C\/JZ1bndn7603ivFNghQ","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Image is not a JPEG.","level":0,"time":"2015-06-15T15:35:22+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"C\/JZ1bndn7603ivFNghQ","remoteAddr":"::1","app":"core","message":"OC_Image->fixOrientation() Orientation: -1","level":0,"time":"2015-06-15T15:35:22+00:00","method":"POST","url":"\/owncloud\/index.php\/avatar\/"}
{"reqId":"jOQOi+1nmhAUeqo08Aex","remoteAddr":"::1","app":"PHP","message":"Array to string conversion at \/srv\/www\/htdocs\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-06-15T15:36:26+00:00","method":"GET","url":"\/owncloud\/index.php\/settings\/admin"}
{"reqId":"9L0YbCz\/IWUjRm4fatta","remoteAddr":"::1","app":"PHP","message":"Automatically populating $HTTP_RAW_POST_DATA is deprecated and will be removed in a future version. To avoid this warning set 'always_populate_raw_post_data' to '-1' in php.ini and use the php:\/\/input stream instead. at Unknown#0","level":3,"time":"2015-06-15T15:36:27+00:00","method":"POST","url":"\/owncloud\/index.php\/displaynames"}
```

#### Browser log
All I could find from Javascript:
```
GET http://localhost/owncloud/index.php/avatar/tmp?requesttoken=nnNR67d89PJjzF0Kf1sFT8KM1cyphV 404 (Not Found)
GET http://localhost/owncloud/index.php/avatar/tmp?requesttoken=nnNR67d89PJjzF0Kf1sFT8KM1cyphV 404 (Not Found)
jquery.min.js?v=b95f2de087b86aa22f6f0288a10176bd:6 POST http://localhost/owncloud/index.php/avatar/cropped 400 (Bad request)x.ajaxTransport.send @ jquery.min.js?v=b95f2de087b86aa22f6f0288a10176bd:6x.extend.ajax @ jquery.min.js?v=b95f2de087b86aa22f6f0288a10176bd:6x.each.x.(anonymous function) @ jquery.min.js?v=b95f2de087b86aa22f6f0288a10176bd:6sendCropData @ personal.js?v=b95f2de087b86aa22f6f0288a10176bd:142(anonymous function) @ personal.js?v=b95f2de087b86aa22f6f0288a10176bd:273x.event.dispatch @ jquery.min.js?v=b95f2de087b86aa22f6f0288a10176bd:5x.event.add.v.handle @ jquery.min.js?v=b95f2de087b86aa22f6f0288a10176bd:5
```

### PR DESCRIPTION
**Pull request by LukasReschke**
This was required by avatars and was broken with https://github.com/owncloud/core/pull/16200

Fixes https://github.com/owncloud/core/issues/16942

@jospoortvliet Please test and film :see_no_evil: 

@DeepDiver1975 @MorrisJobke @blizzz Since you were involved in the PR.

### CODE REVIEW

File: lib/private/avatar.php [hunk 1]
```diff
@@ -90,6 +90,7 @@ public function exists() {
 	 * @return void
 	*/
 	public function set ($data) {
+
```
Comment (LukasReschke) [inline lib/private/avatar.php]: That can probably be removed. Will do once Jenkins passed once to avoid :boom:

**Comment by jospoortvliet** [comment]
@LukasReschke so I modified these two parts in my local install. What happens is that I can indeed pick a new picture - ONCE. The second time I try one, the cropping dialog still shows the first one ;-)

**Comment by LukasReschke** [comment]
Patch confirmed by @jospoortvliet at https://github.com/owncloud/core/issues/16942#issuecomment-112131851

**Comment by jospoortvliet** [comment]
works for me, yes.

**Comment by MorrisJobke** [comment]
Works here too :+1:

**Pull request merged**